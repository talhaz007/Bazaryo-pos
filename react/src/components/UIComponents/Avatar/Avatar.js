import React from "react";
import defaultImage from "../../../Assets/img/placeholder.png";
import "./Avatar.scss";

const Avatar = (props) => {
    const {
        userImage,
        userName,
        size,
        className,
        onClickMain,
        onClickChangePhoto,
    } = props;

    const hoverToEdit = onClickChangePhoto ? "hover-edit" : "";
    const hover = onClickMain ? "hover-pointer" : hoverToEdit;
    return (
        <div
            className={`avatar-wrapper ${size || ""} ${
                className || ""
            } ${hover}`}
            onClick={onClickMain}
        >
            <label htmlFor="myfile">
                <img
                    src={userImage || defaultImage}
                    alt={`${userName || "profile"}`}
                />
                <div className="change-photo-wrapper">
                    <div className="edit-icon" />
                    <div className="edit-text">Change photo</div>
                </div>
            </label>
            <input
                type="file"
                id="myfile"
                name="myfile"
                onChange={onClickChangePhoto}
                style={{ display: "none" }}
            />
        </div>
    );
};

export default Avatar;
