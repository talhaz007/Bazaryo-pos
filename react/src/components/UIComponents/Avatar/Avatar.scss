$font-change-photo-color: #fff;
$background-change-photo-color: #29303b;
div.avatar-wrapper {
    border-radius: 64px;
    position: relative;
    &.xl {
        width: 256px;
        height: 256px;
    }
    &.large {
        width: 128px;
        height: 128px;
    }
    &.medium {
        width: 64px;
        height: 64px;
    }
    &.small {
        width: 32px;
        height: 32px;
    }
    &.ms {
        width: 16px;
        height: 16px;
    }
    img {
        width: 128px;
        height: 128px;
        border-radius: 64px;
    }
    .change-photo-wrapper {
        display: none;
        position: absolute;
        align-items: center;
        justify-content: center;
        cursor: pointer;
        width: 100%;
        .edit-icon {
            background-image: url("../../../Assets/img/placeholder.png");
            background-position: center;
            background-repeat: no-repeat;
            background-size: cover;
            width: 18px;
            height: 18px;
            margin-right: 2px;
        }
        .edit-text {
            text-align: center;
            font-weight: 600;
            font-size: 12px;
            line-height: 16px;
            color: $font-change-photo-color;
        }
    }
    &.hover-edit {
        &:hover {
            background-color: $background-change-photo-color;
            display: flex;
            align-items: center;
            justify-content: center;
            img {
                opacity: 0.3;
            }
            .change-photo-wrapper {
                position: absolute;
                display: flex;
                left: 0px;
                bottom: 38px;
                &:hover {
                    .edit-text {
                        text-decoration: underline;
                    }
                }
            }
        }
    }
    &.hover-pointer {
        &:hover {
            cursor: pointer;
        }
    }
}
