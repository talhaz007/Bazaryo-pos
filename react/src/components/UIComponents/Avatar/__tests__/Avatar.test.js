import React from 'react';
import {shallow} from 'enzyme';
import {
    describe, it, jest, expect
} from '@jest/globals';
import Avatar from '../Avatar';

jest.mock('react-i18next', () => ({
    useTranslation: () => ({t: (key) => key})
}));

describe('Avatar', () => {
    let wrapper;

    it('should be rendered', () => {
        wrapper = shallow(<Avatar/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render an Avatar with image src: http://example.com/image', () => {
        wrapper = shallow(<Avatar userImage="http://example.com/image"/>);
        expect(wrapper.find('img').props().src).toEqual('http://example.com/image');
    });

    it('should render an Avatar with alt attr: Dirk', () => {
        wrapper = shallow(<Avatar userName="Dirk"/>);
        expect(wrapper.find('img').props().alt).toEqual('Dirk');
    });

    it('should render an Avatar in size of medium', () => {
        wrapper = shallow(<Avatar size="medium"/>);
        expect(wrapper.find('.medium')).toHaveLength(1);
    });

    it('should render an Avatar with class name: costumeClassName', () => {
        wrapper = shallow(<Avatar className="costumeClassName"/>);
        expect(wrapper.find('.costumeClassName')).toHaveLength(1);
    });
});
