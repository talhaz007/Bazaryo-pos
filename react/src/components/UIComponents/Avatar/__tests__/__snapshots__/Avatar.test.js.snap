// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should be rendered 1`] = `
<div
  className="avatar-wrapper   "
>
  <img
    alt="profile"
    src="test-file-stub"
  />
  <div
    className="change-photo-wrapper"
  >
    <div
      className="edit-icon"
    />
    <div
      className="edit-text"
    >
      Change photo
    </div>
  </div>
</div>
`;
